src/components/ApolloConnection/ApolloConnection.tsx

```diff
@@ -0,0 +1,136 @@
+import React from "react";
+import {
+    ApolloClient,
+    ApolloProvider,
+    ApolloLink,
+    createHttpLink,
+    InMemoryCache,
+    fromPromise,
+} from "@apollo/client";
+import { setContext } from "@apollo/client/link/context";
+import { getEnvironment } from "../../get-environment";
+import * as SecureStore from "expo-secure-store";
+import { onError } from "@apollo/client/link/error";
+import { REFRESH_TOKEN_MUTATION } from "./refresh-token.mutation";
+import SecureStorageItems from "../../types/SecureStorageItems";
+export type Props = {
+    children: JSX.Element;
+    navigationFn: (name: string, params: any) => void;
+};
+export default function ApolloConnection(props: Props) {
+    // Initialize Apollo Client (Backend)
+    let isRefreshing = false;
+    let pendingRequests: any[] = [];
+
+    const resolvePendingRequests = () => {
+        pendingRequests.map((callback) => callback());
+        pendingRequests = [];
+    };
+
+    const httpLink = createHttpLink({
+        uri: getEnvironment()?.backendUrl,
+        credentials: "include",
+    });
+    const getNewToken = () => {
+        console.log("get new token");
+
+        return client
+            .mutate({ mutation: REFRESH_TOKEN_MUTATION, errorPolicy: "all" })
+            .then((response) => {
+                // extract your accessToken from your response data and return it
+
+                console.log("fetch token");
+                const data = response.data?.refreshToken;
+                if (!data) {
+                    throw new Error("could not get a refresh token");
```

Comment:
How is such an error handled within the UI?

Reply:
This isn't necessary as the user doesn't need to know what happens technically here.
When the error is catched (line 81), the user is redirected to the Login screen, which should be enough for now.

https://github.com/vacatius/mobile-app/blob/ba8acd98fc4c67e02f12d2782f0cd971f7433d00/src/components/ApolloConnection/ApolloConnection.tsx#L81-L89

Reply:
I must've missed that. Thanks!